Use Monaco when creating files in `DebugConfigurationManager`

colin-grant-work opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

This PR fixes #9310 by using Monaco to manipulate the file contents rather than having the `FileService` make the change on the backend only.

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

1. Open a workspace with no `launch.json` files.
2. Create an empty `launch.json` file in some root and have it open.
3. Open the debug widget.
4. Open the dropdown to select configurations and select `Add Configuration`.
5. The file you created should be populated with the default content of a `launch.json` file.
6. Save the file.
7. You should see no error message.
8. It should work regardless of whether the file you create is in a `.vscode` or `.theia` folder.
> The behavior may differ depending on whether you've previously had a `launch.json` in the other folder during the same session.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Colin Grant <colin.grant@ericsson.com>

vince-fugnitto commented on packages/debug/src/browser/debug-configuration-manager.ts:
```diff
@@ -252,28 +252,28 @@ export class DebugConfigurationManager {
     }
 
-    protected async doOpen(model: DebugConfigurationModel): Promise<EditorWidget> {
+    protected async doOpen(model: DebugConfigurationModel): Promise<EditorWidget | undefined> {
```
Should we document this breakage to be safe?
A downstream extension may previously expect `EditorWidget` when calling `doOpen`.

colin-grant-work replied:
Yes, likely best to be safe. I'll push a changelog entry

vince-fugnitto reviewed (approved):
I confirm that the changes successfully fix https://github.com/eclipse-theia/theia/issues/9310 :+1:

colin-grant-work commented:
@vince-fugnitto, as we discussed offline, I've modified the code so that it handles any case in which a `launch.json` does not contain at least `{configurations: []}`. Rather than changing the signatures of the methods, I've thrown an `Error` in case the `PreferenceService` returns no `URI`. As I note in the code, that should never happen here, since the inputs are known. I have added a breaking change notice about swapping out the `FileService` for `MonacoTextModelService`, however.

vince-fugnitto reviewed (approved):
I confirmed that the changes work well :+1: 
- verified that it properly creates the `launch.json` when it does not exist.
- verified that it properly populates the `launch.json` when it does not contain a `configurations` object.
- verified that it properly adds new configurations (through _add configuration_) for an existing `configurations` object.
- works for both `.vscode` and `.theia`.
